src/core/src/main/java/org/apache/jmeter/testelement/TestPlan.java

```diff
@@ -83,11 +86,23 @@ public void prepareForPreCompile()
 
     /**
      * Fetches the functional mode property
-     *
+     * Could be change for no gui test with jmeter property : PROP_FUNCTIONAL_MODE
      * @return functional mode
      */
     public boolean isFunctionalMode() {
-        return getPropertyAsBoolean(FUNCTIONAL_MODE);
+        boolean bFunctionalModeDefault = getPropertyAsBoolean(FUNCTIONAL_MODE);
+        log.debug("bFunctionalModeDefault=" + bFunctionalModeDefault);
+        String sIsNonGui = System.getProperty(org.apache.jmeter.JMeter.JMETER_NON_GUI);
```

Comment:
Same for the *s*. Plus, it is probably not needed, when we use the Method `Boolean.getBoolean(string)`, which combines the lookup of the system property and the conversion to a Boolean.

Reply:
Changed with proposal modification add a private function isNonGui